runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPlugin.java

```diff
@@ -258,10 +261,18 @@ public void onGameTick(GameTick tick)
 		boolean inRegion = client.getLocalPlayer().getWorldLocation().getRegionID() == REGION;
 		if (config.hideButton() && inRegion != buttonAttached)
 		{
+			//Determines state of quest The Depths of Despair when player enters or exits region (quest cannot be
+			//started or ended in the region, so this will not change while player is in region).
+			if (inRegion)
```

Comment:
this won't update if you have hideButton off

Reply:
You're right, thanks for catching that.  I was trying to be slick and I missed that.  I changed it so that check triggers on game tick when in the region.